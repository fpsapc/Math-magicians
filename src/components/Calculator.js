import React from 'react';
import Buttons from './Buttons';

export default function Calculator() {
  return (
    <>
      <Buttons />
    </>
  );
}
