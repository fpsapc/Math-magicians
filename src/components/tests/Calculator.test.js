import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Calculator Page', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return product of two numbers', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByText('2'));
    fireEvent.click(screen.getByText('x'));
    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    const result = screen.getByRole('none');
    expect(result.innerHTML).toBe('6');
  });
});
