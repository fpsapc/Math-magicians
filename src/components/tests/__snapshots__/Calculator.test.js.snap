// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Page renders correctly 1`] = `
<div
  className="calcPage"
>
  <h2
    className="calcText"
  >
    Lets do some math!
  </h2>
  <div
    className="calculator"
  >
    <div
      className="resultBar"
      role="none"
    >
      0
    </div>
    <div
      className="keypad"
    >
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="doubleButton"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="singleButton"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
